contracts/interchain-token-service/InterchainTokenService.sol

```diff
@@ -807,7 +807,7 @@ contract InterchainTokenService is
     function _decodeMetadata(bytes memory metadata) internal pure returns (uint32 version, bytes memory data) {
         data = new bytes(metadata.length - 4);
         assembly {
-            version := shr(224, mload(data))
+            version := shr(224, mload(add(metadata, 32)))
```

Comment:
Please review this change @milapsheth @Foivos @re1ro 
In testing `interchainTransfer` I found that the version would always be zero regardless of what `metadata` I passed in to the function. In the original code `data` is uninitialized and will always be all zero bytes which means that `version` will always revert back to zero regardless of what `metadata` is passed into the decode function, here is my proposed fix. Let me know your thoughts.

Reply:
this is correct